packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -41,19 +46,45 @@ class TestDataObject1 extends DataObject {
 
 	private readonly datastoreKey = "TestDataObject2";
 
-	public createdDataStoreId?: string;
-
 	protected async hasInitialized() {
-		// eslint-disable-next-line @typescript-eslint/no-floating-promises
-		this.init();
+		this.initSync().catch((error) => {});
 	}
 
-	private async init() {
-		const dataObject2 = await rootDataObjectFactory.createInstance(
+	/**
+	 * This function is called during initialization. It creates a data store in summarizer client only if one doesn't
+	 * already exists. The idea behind this is to have data store created during summarization and validate that it
+	 * is handled correctly.
+	 */
+	protected async initSync() {
+		// For non-summarizer (interactive) clients, don't do anything.
+		if (this.context.clientDetails.capabilities.interactive === true) {
+			return;
+		}
+
+		// If the second data store already exists, don't create another one. This ensures that we create data stores
+		// endlessly during summarization.
+		let dataObject2: RootTestDataObject | undefined;
+		const dataObject2Handle = this.root.get<IFluidHandle<RootTestDataObject>>(
+			this.datastoreKey,
+		);
+		if (dataObject2Handle !== undefined) {
+			dataObject2 = await dataObject2Handle.get();
+		}
+		if (dataObject2 !== undefined) {
+			return;
+		}
+
+		const newDataObject = await rootDataObjectFactory.createInstance(
 			this.context.containerRuntime,
 		);
-		this.root.set(this.datastoreKey, dataObject2.handle);
-		this.createdDataStoreId = dataObject2.id;
+		// Store the handle asynchronously to add some delay between data store creation and attaching it. Basically,
+		// the validation during summarization behave differently when data store is in detached, attaching and attached
+		//
+		return Promise.resolve()
```

Comment:
If we're not waiting for a particular other event, is there any difference here adding this `Promise.resolve()` as opposed to just calling it immediately?  Or even worse, any potential race condition?

Reply:
Well, the idea was to add a race condition to imitate what's happening in tasks component but it didn't do anything. So, I would say it doesn't matter what we do here.

Reply:
Changed it to calling it immediately.